### ISSUE TITLE
[WIP]Unable to run axe tool on one of the platform, error occurred message appears 

### ISSUE THREAD

**Issue by gtripathi05** [dequelabs/axe-core#1754]
**Credentials to access the platform**

https://revel.pearson.com/#/start  

Student3.AQA.AXE@mailinator.com /Axeqa123

<img width="956" alt="Axe Issue" src="https://user-images.githubusercontent.com/30168515/62543316-5484fa80-b87b-11e9-8629-cc218d280594.png">

Student4.AQA.AXE@mailinator.com /Axeqa123

Devito_AXEtest_AQA
072219_MG_1036

**Comment by straker** [comment]
Steps to reproduce: after you log into the server, click on the single available course (Devito_AXEtest_AQA). Then click "contents" and navigate to chapter 1.3.

**Comment by AutoSponge** [comment]
At the time of the error, this is the `ruleResult`, the first `result` passed in `Audit.prototype.after`:

```json
{
  "options": {
    "restoreScroll": true,
    "reporter": "v1"
  },
  "command": "rules",
  "parameter": null,
  "context": {
    "initiator": false,
    "page": true,
    "include": [],
    "exclude": []
  }
}
```

Because there's no `id`, it raises an error. All other `results` appear to be instances of `RuleResult`.

**Comment by AutoSponge** [comment]
It looks like the page under test has a listener in an iframe that reflects back our messages (like run) and we don't expect that and so we error.

Possible solution:
```js
axe.utils.respondable(win, 'axe.ping', null, undefined, function(response) {
     if (!response || (response && !response.axe)) return resolve(null);
     clearTimeout(timeout);
```

Needs test page to verify theory.

**Comment by straker** [comment]
A little more context, when we send the command `axe.start` to an iframe, the `data` message we get back in the callback is the exact message we sent: the command `rules`.

**Comment by AkhilaBandiki** [comment]
@AutoSponge

**Comment by AkhilaBandiki** [comment]
@AutoSponge 
Retested the issue mentioned. I am still able to reproduce the issue with latest build. Please refer the below screenshot for the same. Let me know if this is expected? 

**Tested Environment:**
**Axe-pro:** v4.1.0.31917
**Chrome Browser:**  78.0.3904.87
**Server:** http://axe.dequelabs.com
**OS**: MS Windows 10

![image](https://user-images.githubusercontent.com/55178349/69041801-1d93b880-0a16-11ea-9b41-40b988c28901.png)

`Error: Result for unknown rule. You may be running mismatch axe-core versions
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:109812
    at Array.map (<anonymous>)
    at y.after (chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:109749)
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:129252
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:172959
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:143134
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:172959
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:143039
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:142693
    at chrome-extension://hepcmhhcnnkhdbbjgeaelcmfhlmaekop/vendor.bundle.js:46:174874`

**Comment by AutoSponge** [comment]
This was fixed in axe-core. It requires axe-core release to include this fix and then for axe to include the new version.

**Comment by padmavemulapati** [comment]
Verified with the axe and axe-coconut , I am not seeing any issue with axe-coconut but still able to reproduce with axe. I think it requires axe-core release to include this fix and then for axe to include the new version. Wll move it to In progress until we get the fix version

**Comment by schne324** [comment]
@WilcoFiers what needs to happen here? Bump axe version in the extensions?

**Comment by WilcoFiers** [comment]
If it's fixed in coconut, it's fixed as far as axe-core tickets are concerned.
